src/js/chrome/entry.js

```diff
@@ -106,16 +106,16 @@ const App = () => {
       });
   }, []);
 
-  return config ? <RootApp config={config} /> : <div>loading...</div>;
+  return config ? <RootApp config={config} /> : null;
 };
 
 export function rootApp() {
   const { store } = spinUpStore();
-  const pageRoot = document.querySelector('.pf-c-page__drawer');
+  const pageRoot = document.getElementById('chrome-root');
```

Comment:
I wouldn't mount to a new id, but rather reuse the already used if `page` and keep the `body.pug` as is, so we keep the loading there while the JS is loaded and parsed.

Reply:
That is not going to work. The header and the nav along with any other "part" of chrome will have to be part of the same VirtualDOM tree. We need this structure inside the actual React app.

We can add the loading state before the app is mounted later on.

Reply:
We will also no longer have any need for pugs and esi tags because the apps won't have their own HTML templates. But we will need it them until every app is migrated.